src/panels/lovelace/editor/hui-edit-view.ts

```diff
@@ -0,0 +1,276 @@
+              value="${this._id}"
+              .configValue="${"id"}"
+              @value-changed="${this._valueChanged}"
+            ></paper-input>
+            <paper-input
+              label="Title"
+              value="${this._title}"
+              .configValue="${"title"}"
+              @value-changed="${this._valueChanged}"
+            ></paper-input>
+            <paper-input
+              label="Icon"
+              value="${this._icon}"
+              .configValue="${"icon"}"
+              @value-changed="${this._valueChanged}"
+            ></paper-input>
+            <hui-theme-select-editor
+              .hass="${this.hass}"
+              .value="${this._theme}"
+              .configValue="${"theme"}"
+              @theme-changed="${this._valueChanged}"
+            ></hui-theme-select-editor>
+          </div>
+        </paper-dialog-scrollable>
+        <div class="paper-dialog-buttons">
+          <paper-button @click="${this._closeDialog}"
+            >${this.localize("ui.common.cancel")}</paper-button
+          >
+          <paper-button
+            ?disabled="${!this._config || this._saving}"
+            @click="${this._save}"
+          >
+            <paper-spinner
+              ?active="${this._saving}"
+              alt="Saving"
+            ></paper-spinner>
+            ${this.localize("ui.panel.lovelace.editor.edit.save")}</paper-button
+          >
+        </div>
+      </paper-dialog>
+    `;
+  }
+
+  private renderStyle(): TemplateResult {
+    return html`
+      <style>
+        paper-dialog {
+          width: 650px;
+        }
+        paper-button paper-spinner {
+          width: 14px;
+          height: 14px;
+          margin-right: 20px;
+        }
+        paper-spinner {
+          display: none;
+        }
+        paper-spinner[active] {
+          display: block;
+        }
+        .hidden {
+          display: none;
+        }
+        .error {
+          color: #ef5350;
+          border-bottom: 1px solid #ef5350;
+        }
+      </style>
+    `;
+  }
+
+  private _save(): void {
+    this._saving = true;
+    this._updateConfigInBackend();
+  }
+
+  private _saveDone(): void {
+    this._saving = false;
+  }
+
+  private async _resizeDialog(): Promise<void> {
+    await this.updateComplete;
+    fireEvent(this._dialog, "iron-resize");
+  }
+
+  private _closeDialog(): void {
+    fireEvent(this, "cancel-edit-view");
+    this._dialog.close();
+  }
+
+  private async _updateConfigInBackend(): Promise<void> {
+    if (!this._isConfigChanged()) {
+      this._closeDialog();
+      this._saveDone();
+      return;
+    }
+
+    try {
+      if (this.add) {
+        await addView(this.hass!, this._config!, "json");
+      } else {
+        await updateViewConfig(
+          this.hass!,
+          this.viewConfig!.id!,
+          this._config!,
+          "json"
+        );
+      }
+      fireEvent(this, "reload-lovelace");
+      this._closeDialog();
+      this._saveDone();
```

Comment:
Lets just inline this function as the name is really confusing, especially combined with previous 2 lines.